File: backend/app/routers/ai.py
from fastapi import APIRouter
from app.models.ai import (
    CopyGenerationRequest,
    CopyGenerationResponse,
    CategorySuggestionRequest,
    CategorySuggestionResponse
)
from app.services.ai_service import generate_marketing_copy, suggest_category

router = APIRouter(prefix="/api/ai", tags=["AI Marketing & Tools"])

@router.post("/generate-copy", response_model=CopyGenerationResponse)
async def generate_product_copy(payload: CopyGenerationRequest):
    """
    AI Marketing Copywriter:
    Takes 2-3 brief vendor notes and produces a compelling 2-sentence marketing pitch
    optimized for local Sri Lankan buyers.
    """
    return await generate_marketing_copy(title=payload.title, keywords=payload.keywords)

@router.post("/suggest-category", response_model=CategorySuggestionResponse)
async def suggest_product_category(payload: CategorySuggestionRequest):
    """
    Auto-Categorization:
    Recommends the best category tag based on product title and description.
    """
    return await suggest_category(title=payload.title, description=payload.description or "")

File: Rust/src/routers/ai.rs
use axum::{extract::State, routing::post, Json, Router};
use std::sync::Arc;

use crate::{
    models::ai::{
        CategorySuggestionRequest, CategorySuggestionResponse, CopyGenerationRequest,
        CopyGenerationResponse,
    },
    services::ai_service::{generate_marketing_copy, suggest_category},
    AppState,
};

pub fn router() -> Router<Arc<AppState>> {
    Router::new()
        .route("/generate-copy", post(generate_copy_handler))
        .route("/suggest-category", post(suggest_category_handler))
}

async fn generate_copy_handler(
    State(state): State<Arc<AppState>>,
    Json(payload): Json<CopyGenerationRequest>,
) -> Json<CopyGenerationResponse> {
    let client = reqwest::Client::new();
    let resp = generate_marketing_copy(
        &client,
        &state.config.gemini_api_key,
        &payload.title,
        &payload.keywords,
    )
    .await;
    Json(resp)
}

async fn suggest_category_handler(
    Json(payload): Json<CategorySuggestionRequest>,
) -> Json<CategorySuggestionResponse> {
    let resp = suggest_category(
        &payload.title,
        payload.description.as_deref().unwrap_or(""),
    );
    Json(resp)
}
